pmc_heu.cpp

```diff
@@ -108,16 +110,12 @@ int pmc_heu::search_bounds(const pmc_graph& G, std::vector<int>& C_max) {
                 }
 
                 if (mc_cur > mc_prev) {
-                    #pragma omp atomic read acquire
-                    mc_prev = mc;
+                    C.push_back(v);
 
-                    if (mc_cur > mc_prev) {
-                        #pragma omp atomic write release
+                    #pragma omp critical
+                    if (mc_cur > mc) {
                         mc = mc_cur;
 
-                        C.push_back(v);
-
-                        #pragma omp critical
```

Comment:
I was assuming that the atomic access to `mc` acts as a barrier so `C_max` would be updated properly according to `mc`, which is wrong. Here's a counter example sequence:

1. `mc` becomes 3 on iter X. 
2. `mc` becomes 10 on iter Y. 
3. `C_max` is set to the 10-element vector as a result of (2). 
4. `C_max` is set to the 3-element vector as a result of (1).